test/tools/router_check/router.h

```diff
@@ -144,6 +149,10 @@ class RouterCheckTool : Logger::Loggable<Logger::Id::testing> {
 
   bool details_{false};
 
+  bool only_show_failures_{false};
+
+  std::vector<std::pair<std::string, std::vector<std::string>>> tests_;
```

Comment:
Can you add a short comment on what this vector of pairs actually means? It's hard to tell from the signature.